internal/pointer/pointer.go

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package pointer
+
+func Uint32Ptr(n uint32) *uint32 {
+	return &n
+}
+
+func BoolPtr(b bool) *bool {
+	return &b
+}
+
+func StrPtr(s string) *string {
+	return &s
+}
```

Comment:
is it safer to allocate via new() and assign to passed in value instead of returning passed in var ptr ? in the current use where its read only I think should be ok but if we thing about using this pkg, but be better the suggested way what do u think ?

Reply:
That's the same. Golang escape analysis will put the variable on the heap, so the behaviour will be identical (and I think this is way more readable)

Reply:
ok other language won't do this magic "c" so if go can do that is great

Reply:
Yeah with C this would explode.